backend/user-service/utils/mailer.ts

```diff
@@ -0,0 +1,53 @@
+import nodemailer from "nodemailer";
+import dotenv from "dotenv";
+import hbs from "nodemailer-express-handlebars";
+import path from "path";
+import { fileURLToPath } from "url";
+import { Options } from "nodemailer/lib/mailer";
+
+type ExtendedOptions = Options & {
+  template: string;
+  context: Record<string, unknown>;
+};
+
+dotenv.config();
+
+const SERVICE = process.env.SERVICE;
+const USER = process.env.USER;
+const PASS = process.env.PASS;
+
+const transporter = nodemailer.createTransport({
+  service: SERVICE,
+  auth: { user: USER, pass: PASS },
+});
+
+const dirname = fileURLToPath(import.meta.url);
+
+transporter.use(
+  "compile",
+  hbs({
+    viewEngine: {
+      extname: ".hbs",
+      layoutsDir: path.resolve(path.dirname(dirname), "../templates/"),
+      defaultLayout: "",
+    },
+    viewPath: path.resolve(path.dirname(dirname), "../templates/"),
+    extName: ".hbs",
+  })
+);
+
+export const sendAccVerificationMail = async (
+  to: string,
+  subject: string,
+  username: string,
+  verificationLink: string
+) => {
+  const options: ExtendedOptions = {
+    from: USER,
+    to,
+    subject,
+    template: "account-verification",
+    context: { username, verificationLink },
```

Comment:
Do you think it is better if the verification link links to the frontend instead of the api endpoint?

Like maybe the if frontend url format is http://localhost:5173/<email>/<token>, then send a post req to http://localhost:3001/api/users/verify-email/<email>/<token>